app/services/invoices/finalize_batch_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Invoices
+  class FinalizeBatchService < BaseService
+    def initialize(organization:)
+      @organization = organization
+
+      super
+    end
+
+    def call_async
+      Invoices::FinalizeAllJob.perform_later(organization:, invoice_ids: invoices.ids)
```

Comment:
What's the reason to finalize in a loop rather than enqueue single jobs to process in parallel.

For sure one benefit will be to avoid the Sequence error 😄  (see https://github.com/getlago/lago-api/pull/2364) but I'm wondering.

Reply:
to be honest, I was following the pattern we have with `invoices/retry_batch_service` and `invoices/payments/retry_batch_service`, and exactly because we already have the pattern I'd prefer follow it for now, and if we decide to move to parallel, it'd be better to change them alltogether 🙈

Reply:
That's a good reason 😄 👌